lib/schema.json

```diff
@@ -0,0 +1,167 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "nodes": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "opcode": {
+            "type": "string",
+            "enum": [
```

Comment:
You design document says:

pcode names MUST not contain `:` unless:

* They are going to be consumed and renamed by one of the next stages
* They represent platform specific low-level instructions. In this case the
  `<platform-name>:prefix` should be used. `platform-name` should not contain
  any characters except ones defined by the set `[a-zA-Z0-9_\-\.]`

Every node may take several literal inputs, several regular inputs, and
possibly a control input.

We can limit that with schema.